File: C++/ExecCxtWidget.cpp
#include "ExecCxtWidget.h"

//#include <coil/File.h>

#include <QAction>
#include <QLayout>
#include <QMenu>
#include <QMenuBar>
#include <QStatusBar>
#include <QTextEdit>
#include <QFile>
#include <QDataStream>
#include <QFileDialog>
#include <QMessageBox>
#include <QSignalMapper>
#include <QApplication>
#include <QPainter>
#include <QMouseEvent>
#include <QLineEdit>
#include <QComboBox>
#include <QLabel>
#include <QPushButton>
#include <qdebug.h>
#include <QtWidgets>

ExecCxtWidget::ExecCxtWidget(ManagerControl *mgrc, ManagerWidget *mgrWidget, QWidget *parent) : MTabWidget(mgrc, parent)
{

	m_mgrWidget = mgrWidget;
	setGUI("exec_cxt");

	QPushButton *loadECButton = new QPushButton(tc->toUnicode("実行コンテキスト読み込み"));
	subLayouts[subLayouts.size()-1]->addWidget(loadECButton);
	connect(loadECButton, SIGNAL(clicked()), this, SLOT(loadECSlot()));

	QPushButton *setOrderFileButton = new QPushButton(tc->toUnicode("実行順序設定ファイルの読み込み"));
	getWidgetFromName("exec_cxt.periodic.filename")->m_layout->addWidget(setOrderFileButton);
	connect(setOrderFileButton, SIGNAL(clicked()), this, SLOT(setOrderFSlot()));
}

void ExecCxtWidget::loadECSlot()
{
	QString fileName = QFileDialog::getOpenFileName(this,
							tc->toUnicode("開く"), "",
							tr("Dynamic Link Library (*.dll);;Dynamic Link Library (*.so);;All Files (*)"));
	if (fileName.isEmpty())
        return;

	
	std::string ba = fileName.toLocal8Bit();
	
	PathList p = PathList(ba);


	

	QComboBox *wid = dynamic_cast<QComboBox *>(m_mgrWidget->getWidgetFromName("manager.modules.preload")->m_widget);
	if(wid)
	{
		if(wid->findText(tc->toUnicode(p.fname.c_str())) == -1)
		{
			wid->addItem(tc->toUnicode(p.fname.c_str()));
		}
	}
	wid = dynamic_cast<QComboBox *>(m_mgrWidget->getWidgetFromName("manager.modules.load_path")->m_widget);
	
	if(wid)
	{
		if(p.dname == "")
			p.dname = "./" + p.dname;
		if(wid->findText(tc->toUnicode(p.dname.c_str())) == -1)
		{
		
		
			wid->addItem(tc->toUnicode(p.dname.c_str()));
		}
	}

	wid = dynamic_cast<QComboBox *>(getWidgetFromName("exec_cxt.periodic.type")->m_widget);
	
	if(wid)
	{
		if(wid->findText(tc->toUnicode(p.name.c_str())) == -1)
		{
			wid->addItem(tc->toUnicode(p.name.c_str()));
		}
		wid->lineEdit()->setText(tc->toUnicode(p.name.c_str()));
	}
}

void ExecCxtWidget::setOrderFSlot()
{
	QString fileName = QFileDialog::getOpenFileName(this,
							tc->toUnicode("開く"), "",
							tr("Lua File (*.lua);;Configuration File (*.conf);;All Files (*)"));
	if (fileName.isEmpty())
        return;

	QString refileName = QDir().relativeFilePath(fileName);
	std::string ba = refileName.toLocal8Bit();
	QLineEdit *wid = dynamic_cast<QLineEdit *>(getWidgetFromName("exec_cxt.periodic.filename")->m_widget);
	
	wid->setText(tc->toUnicode(ba.c_str()));
}

File: Python/SettingRTCConf/ExecCxtWidget.py
#!/bin/env python
# -*- encoding: utf-8 -*-

##
#   @file .py
#   @brief 



import thread


import optparse
import sys,os,platform
import re
import time
import random
import commands
import math
import imp



import RTC
import OpenRTM_aist

from OpenRTM_aist import CorbaNaming
from OpenRTM_aist import RTObject
from OpenRTM_aist import CorbaConsumer
from omniORB import CORBA
import CosNaming

from PyQt4 import QtCore, QtGui

from MTabWidget import MTabWidget
from ManagerControl import ManagerControl


class ExecCxtWidget(MTabWidget):
    
    def __init__(self, mgrc, mgrWidget, parent=None):
        MTabWidget.__init__(self, mgrc, parent)
        self.setGUI("exec_cxt")

        self.loadECButton = QtGui.QPushButton(u"実行コンテキスト読み込み")
        self.subLayouts[-1].addWidget(self.loadECButton)
        self.loadECButton.clicked.connect(self.loadECSlot)

        self.setOrderFileButton = QtGui.QPushButton(u"実行順序設定ファイルの読み込み")
        self.WidList["exec_cxt.periodic.filename"]["Layout"].addWidget(self.setOrderFileButton)
        self.setOrderFileButton.clicked.connect(self.setOrderFSlot)

        self.mgrWidget = mgrWidget

    def loadECSlot(self):
        fileName = QtGui.QFileDialog.getOpenFileName(self,u"開く","","Python File (*.py);;All Files (*)")

	if fileName.isEmpty():
            return
        ba = str(fileName.toLocal8Bit())
        fname = os.path.basename(ba)
        name, ext = os.path.splitext(fname)
        dname = os.path.dirname(os.path.relpath(ba))
        

        

        wid = self.mgrWidget.WidList["manager.modules.preload"]["Widget"]
        if wid.findText(fname) == -1:
            wid.addItem(fname)

        wid = self.mgrWidget.WidList["manager.modules.load_path"]["Widget"]

        if dname == "":
            dname = "./" + dname
        if wid.findText(dname) == -1:
            wid.addItem(dname)

        wid = self.WidList["exec_cxt.periodic.type"]["Widget"]
        if wid.findText(name) == -1:
            wid.addItem(name)
            wid.lineEdit().setText(name)

    def setOrderFSlot(self):
        fileName = QtGui.QFileDialog.getOpenFileName(self,u"開く","","Python File (*.py);;Configuration File (*.conf);;All Files (*)")
        if fileName.isEmpty():
            return

        ba = str(fileName.toLocal8Bit())
        fname = os.path.relpath(ba)

        wid = self.WidList["exec_cxt.periodic.filename"]["Widget"]
        wid.setText(fname)
